### ISSUE TITLE
Project could be test driven instead of user driven

### ISSUE THREAD

**Issue by smeredith** [TEALS-IntroCS/2nd-semester-introduction-to-computer-science-principles#174]
Instead of interactive programs, I would love to see projects that define function interfaces that students are then required to implement. We could supply tests that call the functions, or just describe them and have the student write them. Grading could be largely automated.

Maybe part 2 of such a project could be gluing the functions together into an interactive program.

This would teach good unit testing practices and habits.

**Comment by lal1029** [comment]
I agree! I've tried out a couple sections where I've provided students with skeletal code and functions that they fill in, in turn, unit tests would be called. A benefit to doing it this way is getting students to focus on the core lesson instead of continually reimplement code that they've already learned. (i.e. input("..."))
